### ISSUE TITLE
[camel-main] Nats native failure, because of missing sun.security.x509.X509Key

### ISSUE THREAD

**Issue by JiriOndrusek** [apache/camel-quarkus#5233]
### Bug description

After upgrade of JNats to 2.16.14 (in Camel), there is a native error about unknown type `sun.security.x509.X509Key` when parsing `EdDSAEngine`

I added a module containing x509Key into test module quarkus execution by

```
    <properties>
        <!-- his is required for the native execution (probably also for JVM).
            EdDSAEngine requires sun.security.x509.X509Ke, which is no longer part if the java (since J16)
             (Similar approach was used in kudu extension) -->
        <opens>java.base/sun.security.x509=ALL-UNNAMED</opens>
        <argLine>--add-opens ${opens}</argLine>
        <quarkus.native.additional-build-args>-J--add-opens=${opens}</quarkus.native.additional-build-args>
    </properties>
```
which fixes the error.

----

I'm not sure why upgrade of the dependency is causing this error.

If addition of the module is the right solution, documentation should be updated to cover thus requirement.
(Another solution might exist. It might be possible to remove `sun.security.x509.X509Key` part from `EdDSAEngine` by using @Substitute - but I didn't try it)

**Comment by jamesnetherton** [comment]
> Another solution might exist. It might be possible to remove sun.security.x509.X509Key part from EdDSAEngine

Yes, I pushed a commit to do that. Replicating this unreleased change:

https://github.com/str4d/ed25519-java/commit/35c34a549cc933dc2d1d23ad4bfa88187fe77e7a

We can remove it if we ever get a 0.3.1 release of `net.i2p.crypto:eddsa`. But that project seem inactive...

**Comment by gansheer** [comment]
@JiriOndrusek do you mind if I look into it ? I am working on Nats with https://github.com/apache/camel-quarkus/issues/7771 and https://github.com/apache/camel-quarkus/issues/2212.

**Comment by JiriOndrusek** [comment]
Hi, sure, please look at this one! (@gansheer )

**Comment by gansheer** [comment]
Closing it since eddsa related code has been removed in https://github.com/apache/camel-quarkus/commit/2cc44211fc7, IMO this is no longer relevant.
@JiriOndrusek Don't hesitate to reopen it if I missed something.
